osu.Game/Screens/Play/KeyCounterDisplay.cs

```diff
@@ -100,7 +101,9 @@ public Color4 KeyUpTextColor
             }
         }
 
-        private void updateVisibility() => this.FadeTo(Visible.Value || configVisibility.Value ? 1 : 0, duration);
+        private void updateVisibility() =>
+            // Change visibility of all key counters internally to isolate from showing them by fading in this container.
+            Children.ForEach(k => k.FadeTo(Visible.Value || ConfigVisibility.Value ? 1 : 0, duration));
```

Comment:
make a container rather than applying this to individual children.

Reply:
Also I'm really confused by the logic - isn't
```csharp
private void updateVisibility() => this.FadeTo(Visible.Value && configVisibility.Value ? 1 : 0, duration);
```
what we really want? How's this change fixing the problem?

Reply:
@peppy I was unsure at first, that way would cause a huge change to the `KeyCounterDisplay` (inherit `CompositeDrawable` and make the fill flow private), but if you think it's better that way, will do.

Reply:
yes, this is how it should be done. it's only a few lines of change.